app/models/user.rb

```diff
@@ -98,6 +98,7 @@ class User < Principal
            inverse_of: :user,
            dependent: :destroy
 
+  has_many :documents, foreign_key: :author_id, dependent: :destroy, inverse_of: :author
```

Comment:
From discussion with @brunopagno - the document is owned by the project, not exclusively by the author. When the author is deleted, the reference should be replaced with the "Deleted User" as in https://github.com/opf/openproject/pull/19999/files#diff-9e11cdc5737091941d4a9f2f454a194e846b9964bdc1c5e2f66974d528bc5a81

```suggestion
  has_many :documents, foreign_key: :author_id, dependent: :nullify, inverse_of: :author
```